packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -0,0 +1,1121 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'button.dart';
+import 'debug.dart';
+import 'theme.dart';
+import 'theme_data.dart';
+import 'toggle_buttons_theme.dart';
+
+/// A horizontal set of toggle buttons.
+///
+/// The list of [children] are laid out in a row. The state of each button
+/// is controlled by [isSelected], which is a list of bools that determine
+/// if a button is in an active, disabled, or selected state. They are both
+/// correlated by their index in the list. The length of [isSelected] has to
+/// match the length of the [children] list.
+///
+/// ## Customizing toggle buttons
+/// The toggle buttons are designed to be configurable, meaning the actions
```

Comment:
Maybe too many words for not enough content. I assume that what you're trying to say is that each toggle's behavior is defined by its [onPressed] callback which can update the [isSelected] list however it wants to.